src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_item.c

```diff
@@ -1,85 +1,1000 @@
+    150, // SLOT_POWDER_KEG
+    150, // SLOT_PICTO_BOX
 };
 
-u8 D_8082B47C[] = {
-    0x35, 0x33, 0x34, 0x32, 0xFF,
-};
+void KaleidoScope_DrawAmmoCount(PauseContext* pauseCtx, GraphicsContext* gfxCtx, s16 item, u16 ammoIndex) {
+    s16 ammoUpperDigit;
+    s16 ammo;
 
-s16 D_8082B484[] = {
-    0x0294,
-    0x0384,
-    0x0474,
-    0x0000,
-};
+    OPEN_DISPS(gfxCtx);
+
+    if (item == ITEM_PICTO_BOX) {
+        if (!CHECK_QUEST_ITEM(QUEST_PICTOGRAPH)) {
+            ammo = 0;
+        } else {
+            ammo = 1;
+        }
+    } else {
+        ammo = AMMO(item);
+    }
+
+    gDPPipeSync(POLY_OPA_DISP++);
+
+    if (!gPlayerFormSlotRestrictions[((void)0, gSaveContext.save.playerForm)][SLOT(item)]) {
+        // Ammo item is restricted
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 100, 100, 100, pauseCtx->alpha);
+    } else {
+        // Default ammo
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+        if (ammo == 0) {
+            // Out of ammo
+            gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 130, 130, 130, pauseCtx->alpha);
+        } else if (((item == ITEM_BOMB) && (AMMO(item) == CUR_CAPACITY(UPG_BOMB_BAG))) ||
+                   ((item == ITEM_BOW) && (AMMO(item) == CUR_CAPACITY(UPG_QUIVER))) ||
+                   ((item == ITEM_STICK) && (AMMO(item) == CUR_CAPACITY(UPG_STICKS))) ||
+                   ((item == ITEM_NUT) && (AMMO(item) == CUR_CAPACITY(UPG_NUTS))) ||
+                   ((item == ITEM_BOMBCHU) && (AMMO(item) == CUR_CAPACITY(UPG_BOMB_BAG))) ||
+                   ((item == ITEM_POWDER_KEG) && (ammo == 1)) || ((item == ITEM_PICTO_BOX) && (ammo == 1)) ||
+                   ((item == ITEM_MAGIC_BEANS) && (ammo == 20))) {
+            // Ammo at capacity
+            gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 120, 255, 0, pauseCtx->alpha);
+        }
+    }
+
+    // Separate ammo into upper and lower digits
+    for (ammoUpperDigit = 0; ammo >= 10; ammoUpperDigit++) {
+        ammo -= 10;
+    }
+
+    gDPPipeSync(POLY_OPA_DISP++);
+
+    // Draw upper digit
+    if (ammoUpperDigit != 0) {
+        POLY_OPA_DISP = func_8010CD98(POLY_OPA_DISP, ((u8*)gAmmoDigit0Tex + (8 * 8 * ammoUpperDigit)), 8, 8,
+                                      sAmmoRectLeft[ammoIndex], sAmmoRectHeight[ammoIndex], 8, 8, 1 << 10, 1 << 10);
+    }
+
+    // Draw lower digit
+    POLY_OPA_DISP = func_8010CD98(POLY_OPA_DISP, ((u8*)gAmmoDigit0Tex + (8 * 8 * ammo)), 8, 8,
+                                  sAmmoRectLeft[ammoIndex] + 6, sAmmoRectHeight[ammoIndex], 8, 8, 1 << 10, 1 << 10);
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void KaleidoScope_SetCursorVtx(PauseContext* pauseCtx, u16 index, Vtx* vtx) {
+    pauseCtx->cursorVtx[0].v.ob[0] = vtx[index].v.ob[0];
+    pauseCtx->cursorVtx[0].v.ob[1] = vtx[index].v.ob[1];
+}
+
+s16 sMagicArrowEffectsR[] = { 255, 100, 255 };
+s16 sMagicArrowEffectsG[] = { 0, 100, 255 };
+s16 sMagicArrowEffectsB[] = { 0, 255, 100 };
+void KaleidoScope_DrawItemSelect(PlayState* play) {
```

Comment:
```suggestion

void KaleidoScope_DrawItemSelect(PlayState* play) {
```
although I'm guessing these will end up as IFS.